src/views/ContentReferenceWidget.vue

```diff
@@ -66,13 +68,102 @@ export default {
 		},
 	},
 
+	data() {
+		return {
+			element: {
+				...this.richObject,
+			},
```

Comment:
It seems like when I don't do this and use `richObject` directly, it's not reactive: it actually updates the properties, but doesn't reflect it in the DOM. Also I thought it's not safe to modify props from child components anyway? So I think what I was doing here is storing the prop in the component's `data`, so that it becomes reactive and it can safely be modified.

This may be a stupid question by now, but I'm pretty sure Tables uses Vue 2 still, yeah? If so, I found the [One-Way Data Flow](https://v2.vuejs.org/v2/guide/components-props#One-Way-Data-Flow) documentation provides some insight into this.

Let me know your thoughts

Reply:
It seems like `SmartPicker.vue` actually is what is using the `ContentReferenceWidget.vue` component, so maybe we can instead have the widget emit an event or something and passes the new/edited row up to the parent, so it can then be reflected using the `richObject` prop?

Reply:
I think my comment was quite vague. Reactivity can be a lot of things. I was referring to `element` not getting updated when `richObject` changes. So the one way data flow down into the component being interrupted. This might be what you want if you want to change the data inside the component so it becomes different from the prop. That's often the case in forms. You seed the form with the initial content and then the user can edit them independently before submitting it. Maybe that's what you are doing here?

> Tables uses Vue 2 still, yeah?
Yes.

I don't really understand the use of the view here. Maybe we can briefly talk about that later. That's probably easier.

Reply:
> You seed the form with the initial content and then the user can edit them independently before submitting it. Maybe that's what you are doing here?

Yeah basically. I've since changed it to not copy the entire `richObject` prop, but instead just the rows, because that is the only data which changes. The goal was to make it this way so that if you edit or create a row from the widget, the new or edited row will be reflected immediately in the widget without requiring a page reload.